app/controllers/users_controller.rb

```diff
@@ -22,8 +29,50 @@ def create
     end
   end
 
+  # GET /users/:id/edit
+  def edit; end
+
+  # PATCH/PUT /users/:id
+  def update
+    if @user.update user_params
+      flash[:success] = t(".updated")
+      redirect_to @user
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /users/:id
+  def destroy
+    if @user.destroy
+      flash[:success] = t(".deleted")
+    else
+      flash[:danger] = t(".delete_failed")
+    end
+    redirect_to users_path
+  end
+
   private
 
+  def logged_in_user
+    return if logged_in?
+
+    store_location
+    flash[:danger] = t(".please_log_in")
+    redirect_to login_url
+  end
+
+  def correct_user
+    return if current_user? @user
+
+    flash[:error] = t(".cannot_edit")
+    redirect_to root_url
+  end
+
+  def admin_user
+    redirect_to_root_path unless current_user.admin?
```

Comment:
```suggestion
    return if current_user.admin?
    
    flash ...
    redirect_to_root_path
```
Dùng return if, bổ sung thêm flash messages báo lỗi nữa e nhé